core/domain/stats_jobs_one_off.py

```diff
@@ -247,64 +246,67 @@ def prepare_map(cls, item):
+                'state_name': item.state_name,
+                'id': item.id,
+                'created_on': python_utils.UNICODE(item.created_on),
+                'session_id': item.session_id
+            }
         elif isinstance(
                 item, stats_models.StartExplorationEventLogEntryModel):
-            return (
-                item.exploration_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_START_EXPLORATION,
-                    'version': item.exploration_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
+            event_dict = {
+                'event_type': feconf.EVENT_TYPE_START_EXPLORATION,
+                'version': item.exploration_version,
+                'state_name': item.state_name,
+                'id': item.id,
+                'created_on': python_utils.UNICODE(item.created_on),
+                'session_id': item.session_id
+            }
         elif isinstance(
                 item, stats_models.ExplorationActualStartEventLogEntryModel):
-            return (
-                item.exp_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION,
-                    'version': item.exp_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
+            event_dict = {
+                'event_type': feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION,
+                'version': item.exp_version,
+                'state_name': item.state_name,
+                'id': item.id,
+                'created_on': python_utils.UNICODE(item.created_on),
+                'session_id': item.session_id
+            }
         elif isinstance(
                 item, stats_models.CompleteExplorationEventLogEntryModel):
-            return (
-                item.exploration_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_COMPLETE_EXPLORATION,
-                    'version': item.exploration_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        # pylint: enable=too-many-return-statements
+            event_dict = {
+                'event_type': feconf.EVENT_TYPE_COMPLETE_EXPLORATION,
+                'version': item.exploration_version,
+                'state_name': item.state_name,
+                'id': item.id,
+                'created_on': python_utils.UNICODE(item.created_on),
+                'session_id': item.session_id
+            }
+
+        if (
```

Comment:
No -- don't have two parallel if/elses, please, otherwise it's hard to maintain. Treat each case exactly once, just like it was done before. 

The point is that you can group some of the cases together. I know the dicts differ slightly but you can decide the event type based on the specific instance type, construct the dict, and return it.

Reply:
Not just `event_type` but every key differ in each instance type because they get their values from instance `item`, therefore we need at least two if/elses. If I use nested if/else, would that work?

Reply:
You do not need to combine all the groups into a single thing. I gave you a suggested grouping in a previous comment, which tries to match dicts of the same format.

Reply:
I don't mean that, Let's suppose we have this dictionary, then the values corresponding to each key will change with every instance type, so we can combine all the groups into a single thing, we can be easily done. But for event_type type even the variable change every time, so to assign that value to event_type we need a nested if to check the instance type and assign the needed value.

```
{
                    'event_type': feconf.EVENT_TYPE_ANSWER_SUBMITTED,
                    'version': item.exp_version,
                    'state_name': item.state_name,
                    'id': item.id,
                    'created_on': python_utils.UNICODE(item.created_on),
                    'is_feedback_useful': item.is_feedback_useful
                })
```

Reply:
Yes. I agree with that, or alternatively you can create a dict mapping the instance type to the event type and just read off of it.

Reply:
I've made the changes, PTAL if they are alright.